akd/src/directory.rs

```diff
@@ -529,6 +524,17 @@ where
                 .await?;
         }
 
+        let mut update_proofs = Vec::<UpdateProof>::new();
```

Comment:
Hmm, perhaps to avoid a future attempt to optimize this, can we add a comment to this codeblock saying that it should only happen after the preloading is finished?

```
// The creation of update proofs should happen only after the preload operation (to prevent cache misses).
```

Reply:
added comment